starlette/config.py

```diff
@@ -78,6 +78,8 @@ def _read_file(self, file_name: str) -> typing.Dict[str, str]:
                 line = line.strip()
                 if "=" in line and not line.startswith("#"):
                     key, value = line.split("=", 1)
+                    if "#" in value:
+                        value = value.split("#", 1)[0]
```

Comment:
This will fail for `SOME_URL=http://example.com#fragment`.

Reply:
Indeed, we'd need a smarter approach here.

Reply:
@blueyed Thanks for pointing out the issue might occur. I will think about a smarter approach here.